panel/widgets/input.py

```diff
@@ -57,6 +58,9 @@ class TextInput(Widget):
     description = param.String(default=None, doc="""
         An HTML string describing the function of this component.""")
 
+    enter_pressed = param.Integer(default=0, readonly=True, doc="""
```

Comment:
Would have this as a `param.Action` and sync the event as a Bokeh Event rather than a counter. May have to do something like `ChatAreaInput` though and send the current value_input in the event since events will be processed faster than the `value_input` property change.